p2p/node.go

```diff
@@ -85,14 +89,14 @@ var _ p2ptest.NodeMakerFunc = NewTestableQriNode
 
 // NewTestableQriNode creates a new node, as a TestablePeerNode, usable by testing utilities.
 func NewTestableQriNode(r repo.Repo, p2pconf *config.P2P) (p2ptest.TestablePeerNode, error) {
-	return NewQriNode(r, p2pconf)
+	return NewQriNode(r, p2pconf, &event.NilBus)
 }
 
 // NewQriNode creates a new node from a configuration. To get a fully connected
 // node that's searching for peers call:
 // n, _ := NewQriNode(r, cfg)
 // n.GoOnline()
-func NewQriNode(r repo.Repo, p2pconf *config.P2P) (node *QriNode, err error) {
+func NewQriNode(r repo.Repo, p2pconf *config.P2P, pub event.Publisher) (node *QriNode, err error) {
```

Comment:
packages that publish events must provide tests proving they work. Let's rework the antiquated `TestNewNode` function in `node_test.go` to look like this, which also tests the `ETP2PGoneOnline` event:

```go
package p2p

import (
	"context"
	"testing"

	ma "github.com/multiformats/go-multiaddr"
	"github.com/qri-io/qri/config"
	cfgtest "github.com/qri-io/qri/config/test"
	"github.com/qri-io/qri/event"
	"github.com/qri-io/qri/repo/profile"
	"github.com/qri-io/qri/repo/test"
)

func TestNewNode(t *testing.T) {
	ctx, cancel := context.WithCancel(context.Background())
	defer cancel()

	info := cfgtest.GetTestPeerInfo(0)
	r, err := test.NewTestRepoFromProfileID(profile.IDFromPeerID(info.PeerID), 0, -1)
	if err != nil {
		t.Errorf("error creating test repo: %s", err.Error())
		return
	}

	bus := event.NewBus(ctx)

	eventFired := make(chan struct{}, 1)
	bus.Subscribe(func(_ context.Context, typ event.Type, payload interface{}) error {
		if typ == event.ETP2PGoneOnline {
			if _, ok := payload.([]ma.Multiaddr); !ok {
				t.Errorf("expected %q event to have a payload of []multiaddr.Multiaddr, got: %T", event.ETP2PGoneOnline, payload)
			}
			eventFired <- struct{}{}
		}
		return nil
	}, event.ETP2PGoneOnline)

	p2pconf := config.DefaultP2PForTesting()
	node, err := NewQriNode(r, p2pconf, bus)
	if err != nil {
		t.Fatalf("error creating qri node: %s", err.Error())
		return
	}
	if node.Online {
		t.Errorf("default node online flag should be false")
	}
	if err := node.GoOnline(ctx); err != nil {
		t.Error(err.Error())
	}
	if !node.Online {
		t.Errorf("online should equal true")
	}

	<-eventFired
}
```

Reply:
Dope! I actually have the re-worked version of this for all the events we've been firing on the rebased `refactor_testground` branch, but yes totally makes sense to have it up and running here.